tests/main.tf

```diff
@@ -3,7 +3,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "~> 3.45"
+      version = "~> 4.0"
```

Comment:
There are a lot of changes in this [diff](https://github.com/hashicorp/terraform-provider-azurerm/compare/v3.45.0...v4.0.0), I recommend we look a bit at this first.